### DISCUSSION TITLE
Disable access on port 3000

### DISCUSSION THREAD

**Discussion by hywax** [Dokploy/dokploy#879] [Q&A]
After domain binding I would like to disable access directly by ip. How can I do that? I have not found such a point in the documentation

**Comment by Siumauricio** [comment]
@hywax you can use the firewall and restrict the port 3000

**Comment by Ikram-Maulana** [comment]
@Siumauricio Then what if the condition when the vps I use does not have a firewall that is managed for free so I have to configure it through ufw. But after I closed the port and only opened for port 22, 80 and 443, ufw does not work with docker so port 3000 is still accessible. The result is that the domain and ip:port are both accessible, not just the domain that should be accessible. do you have any suggestions for me?

**Comment by QThans** [comment]
I have same question. After check, I consider it maybe docker and ufw are conflcting for port control. It need to change rules of iptables. Do you have better way to control it by using ufw directly? Also for all compose services have same question. @Siumauricio

**Comment by jannismilz** [comment]
Is there an update on this?

Are we forced to use the firewall?
We probably can't recreate the container without exposing the port, right? Because after each update it will be exposed again?

**Comment by schummar** [comment]
You can remove the published port with
```
docker service update dokploy --publish-rm published=3000,target=3000,mode=host
```
This will update the swarm service and automatically recreate the container.

I have not had the opportunity to see whether an update resets it, but probably not.

**Comment by revant** [comment]
I did this https://github.com/Dokploy/dokploy/discussions/1254

**Comment by heyJordanParker** [comment]
@schummar I just tested this. Updates do not reset it. This is the way.

**Comment by OzkrRouj** [comment]
@schummar I used this command. It works for me. Thanks!!!

**Comment by Siumauricio** [comment]
We added docs for removing the 3000 port in your dokploy instance https://docs.dokploy.com/docs/core/installation#disable-access-via-ipport-optional-but-recommended

**Reply by tonydehnke** [reply]
> We added docs for removing the 3000 port in your dokploy instance https://docs.dokploy.com/docs/core/installation#disable-access-via-ipport-optional-but-recommended

Heads up @Siumauricio  the link to the secure recommendations docs is broken.  on that page, below the command line. 

![image](https://github.com/user-attachments/assets/a4326e3b-a116-4eda-9871-6d03dd2e4373)

**Comment by aabidk20** [comment]
Hey @Siumauricio, is there a way to use 127.0.0.1 instead of 0.0.0.0 as host? I don't want to remove port 3000 completely as it can allow me to access the server via tailscale

**Reply by Guibzs** [reply]
Did you find a way to do it ? I'm wondering for the same thing.
